src/main/java/com/endava/internship/service/UserServiceImpl.java

```diff
@@ -2,68 +2,99 @@
 
 import com.endava.internship.domain.Privilege;
 import com.endava.internship.domain.User;
-import com.endava.internship.service.UserService;
 
+import java.util.AbstractMap;
+import java.util.Arrays;
+import java.util.Comparator;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.function.Function;
 import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
 public class UserServiceImpl implements UserService {
 
     @Override
     public List<String> getFirstNamesReverseSorted(List<User> users) {
-        throw new UnsupportedOperationException("Not implemented");
+        return users.stream().
+                map(User::getFirstName).
+                sorted(Comparator.reverseOrder()).
+                collect(Collectors.toList());
     }
 
     @Override
     public List<User> sortByAgeDescAndNameAsc(final List<User> users) {
-        throw new UnsupportedOperationException("Not implemented");
+        return users.stream().
+                sorted(Comparator.comparing(User::getFirstName)).
```

Comment:
you can use construction like 
`comparing(age).reversed().thenComparing(name)`

Reply:
fixed